### PR TITLE
Zunion

### PR DESCRIPTION
**Pull request by AvitalFineRedis**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `$ tox` pass with this change (including linting)?
- [x] Do the CI tests pass with this change (enable it first in your forked repo and wait for the github action build to finish)?
- [x] Is the new or changed code fully tested?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?

_NOTE: these things are not required to open a PR and can be done
afterwards / while the PR is open._

### Description of change

_Please provide a description of the change here._

### CODE REVIEW

**Comment by AvitalFineRedis** [comment]
waiting for ZINTER to be merged

**Comment by chayim** [comment]
@AvitalFineRedis It looks like your zunion changes have were based off [the zinter PR](https://github.com/andymccurdy/redis-py/pull/1520). Mind breaking them up?

**Comment by AvitalFineRedis** [comment]
@chayim I needed the changes in the aggregation function for the ZUNION support.
After ZINTER will be merged, they will break up and you will see only the changes of ZUNION.

File: redis/client.py
Comment (chayim) [inline redis/client.py:3265]: What about the weights? https://redis.io/commands/ZUNION

**Review by chayim — CHANGES_REQUESTED** [review]
I think we need support for weights.

File: redis/client.py
Comment (AvitalFineRedis) [reply redis/client.py:3265]: @chayim You can see in the test.
`_zaggregate` supports weights (can be provided as a dictionary with the `keys` argument).

Comment (AvitalFineRedis) [reply redis/client.py:3265]: I will add it to the docstring

**Review by chayim — APPROVED** [review]

**Pull request merged**
